src/main/java/frc/robot/commands/Shoot.java

```diff
@@ -0,0 +1,103 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.subsystems.gpm_subsystem.Outtake;
+import frc.robot.subsystems.gpm_subsystem.Wrist;
+
+/**
+ * Shoots on the move.
+ */
+public class Shoot extends Command {
+
+    private static final Pose3d SPEAKER_POSE = new Pose3d(0, 0, 2.055, new Rotation3d());
+    private static final double SHOOTER_HEIGHT = 0;
+
+    private final Outtake outtake;
+    private final Wrist wrist;
+    private final Drivetrain drivetrain;
+
+    private Pose3d displacement;
+    private double v_rx;
+    private double v_ry;
+
+    public Shoot(Outtake outtake, Wrist wrist, Drivetrain drivetrain) {
+        this.outtake = outtake;
+        this.wrist = wrist;
+        this.drivetrain = drivetrain;
+        addRequirements(outtake);
+    }
+
+    @Override
+    public void initialize() {
+        drivetrain.setIsShooting(true);
+    }
+
+    @Override
+    public void execute() {
+        // Set displacement to speaker
+        displacement =
+                new Pose3d(
+                        drivetrain.getPose().getX(),
+                        drivetrain.getPose().getY(),
+                        SHOOTER_HEIGHT,
+                        new Rotation3d(
+                                drivetrain.getPose().getRotation().getCos(),
+                                wrist.getAngle(),
+                                drivetrain.getPose().getRotation().getSin() // TODO: is it cos->sin or sin->cos
```

Comment:
If you want this to be the shooter's pose, it should be neither sin nor cos. The constructor of Rotation3d takes a roll, pitch, and yaw, so the roll should be 0 and the yaw should be the drivetrain's rotation. Currently, this will always have a roll and yaw between -1 radian and 1 radian.